### PR TITLE
Panel edit: Run queries when time range changes in table view

### LINKED ISSUE #52613: Dashboard: Time Range object doesn't update in table view
**What happened**: The time range doesn't update in the data source query hook for a custom data source plugin when in table view

**What you expected to happen**: I expect to time range object to update even when in the table view.

**How to reproduce it (as minimally and precisely as possible)**: console log the time range object when in table view and change the time range, it will keep passing the same time range when you first switched to table view, if you turn off table view then time range will pass correctly.

**Anything else we need to know?**:

**Environment**: docker
- Grafana version: v9.0.3 (023f9251a9)
- Data source type & version: custom data source plugin
- OS Grafana is installed on: running in Docker: on Debian OS
- User OS & Browser: Mac: Chrome
- Grafana plugins: none, other than frontend Datasource template

grafana-time-bug

### PR DESCRIPTION
**Pull request by axelavargas**
**What this PR does / why we need it**:

In `PanelEditorTableView` when TimeRange was changed, panel queries were not receiving the update.

https://user-images.githubusercontent.com/239999/186858825-7523a73e-b509-43ea-b983-4c62c4fbd122.mp4

**Which issue(s) this PR fixes**:

Fixes #52613

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/dashboard/components/PanelEditor/PanelEditorTableView.tsx [hunk 1]
```diff
@@ -30,9 +30,9 @@ export function PanelEditorTableView({ width, height, panel, dashboard }: Props)
   // Subscribe to panel event
   useEffect(() => {
     const timeSrv = getTimeSrv();
-    const timeData = applyPanelTimeOverrides(panel, timeSrv.timeRange());
-
+    let timeData = applyPanelTimeOverrides(panel, timeSrv.timeRange());
     const sub = panel.events.subscribe(RefreshEvent, () => {
+      timeData = applyPanelTimeOverrides(panel, timeSrv.timeRange());
       panel.runAllPanelQueries(dashboard.id, dashboard.getTimezone(), timeData, width);
     });
```
Comment (dprokop) [inline public/app/features/dashboard/components/PanelEditor/PanelEditorTableView.tsx:37]: I think this change makes sense, but I suggest calculating the time data only in the subscription, as the first calculation is not used at all:

```suggestion
    const sub = panel.events.subscribe(RefreshEvent, () => {
      const timeData = applyPanelTimeOverrides(panel, timeSrv.timeRange());
      panel.runAllPanelQueries(dashboard.id, dashboard.getTimezone(), timeData, width);
    });
```

File: public/app/features/dashboard/components/PanelEditor/PanelEditorTableView.test.tsx [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    ...timeRange,
+  })),
+}));
+
+// Mock timeSrv singleton instance
+const timeSrvMock = new TimeSrvStub();
+jest.mock('../../services/TimeSrv', () => ({
+  ...jest.requireActual('../../services/TimeSrv'),
+  getTimeSrv: jest.fn(() => {
+    return timeSrvMock;
```
Comment (axelavargas) [inline public/app/features/dashboard/components/PanelEditor/PanelEditorTableView.test.tsx:36]: 🤔 I am not sure if this is the best way to mock `getTimeSrv`

Comment (dprokop) [reply public/app/features/dashboard/components/PanelEditor/PanelEditorTableView.test.tsx:36]: @axelavargas I think a simpler way would be to just use the service setter (`setTimeSrv` from 
 `app/features/dashboard/services/TimeSrv`), and just do the following in `beforeAll`:

```
setTimeSrv(timeSrvMock)
```

Comment (axelavargas) [reply public/app/features/dashboard/components/PanelEditor/PanelEditorTableView.test.tsx]: yes! that's a way better approach, thank you, Dominik 🙏🏾 .  I just commit the suggestion 🙌🏾

**Review by dprokop — APPROVED** [review]
👏

**Pull request merged**